lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -67,11 +33,10 @@ defmodule LambdaEthereumConsensus.ForkChoice do
     :telemetry.execute([:sync, :on_block], %{slot: head_slot})
 
     persist_store(store)
-    {:ok, store}
   end
 
-  @impl GenServer
-  def handle_cast({:on_block, %BlockInfo{} = block_info, from}, _store) do
+  @spec on_block(BlockInfo.t(), pid()) :: :ok | :error
+  def on_block(%BlockInfo{} = block_info, from \\ self()) do
```

Comment:
If this is no longer a GenServer cast, it's no longer necessary to cast back to the caller using `from`. Let's just return the result instead and process it directly in PendingBlocks.

Reply:
You are right. Done!